assets/js/src/segments/dynamic/dynamic_segments_filters/email_statistics_opens.tsx

```diff
@@ -42,12 +43,14 @@ export const EmailOpenStatisticsFields: React.FunctionComponent<Props> = ({ filt
   useEffect(() => {
     if ((segment.operator !== AnyValueTypes.ANY)
         && (segment.operator !== AnyValueTypes.ALL)
-        && (segment.operator !== AnyValueTypes.NONE)
+        && ((segment.operator !== AnyValueTypes.NONE)
+          || (segment.action === EmailActionTypes.MACHINE_OPENED)
```

Comment:
This condition seems a bit odd to me. Are you sure it is correct? Could you please leave a code comment on why do we need it this way? Or maybe even refactor it to a separate method and make it more readable...

Reply:
This is needed because machine-open doesn't support `none of`. But a good point it is not readable. I'll refactor it.